msprime/ancestry.py

```diff
@@ -1146,6 +1146,128 @@ def sim_ancestry(
     )
 
 
+def continue_simulation(
+    ts,
+    time,
+    sample_size=None,
+    continue_nodes=None,
+    **kwargs,
+):
+    """
+    Returns a full tree sequence, by continuing the simulation using msprime
+    after an earlier simulation has been performed. The msprime simulation and slim
+    simulation are joined using :meth:`tskit.tables.TableCollection.union()`.
+
+    This method executes code from the pyslim tutorial  'Following up with more
+    coalescent simulation'
+
+    ``sample_size`` defaults to the number of samples present in the input tree sequence.
+
+    In general, all parameters  excepting ``samples`` are whatever the defaults
+    of ``msprime.simulate`` are; this includes recombination rate, so that if neither
+    ``recombination_rate`` or a ``recombination_map`` are provided, there will be
+    *no* recombination.
+
+    Note that ``Ne`` defaults to ``1.0``; you may want to set it explicitly
+    using kwargs.
+
+    :type ts: :class`tskit.TreeSequence`
+    :param ts: tree sequence used to continue siumlation, moving forward in time
+    :param float time: The desired period of time to continue simulation in msprime,
+        in units of generations
+    :param int sample_size: Sample size for :meth:`msprime.simulate`,
+        in units of haploid samples,
+    :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+    :type nodes: :class`numpy.ndarray`
+    :param continue_nodes: node IDs in ts to match up to the lineages in the continued
+        simulation, defaults to the samples
+    :return: The :class:`tskit.TreeSequence` object resulting from the
+        union of the input tree equence and the continued simulatoin
+    :rtype: :class:`tskit.TreeSequence`
+    """
+    if continue_nodes is None:
+        continue_nodes = ts.samples()
+    if sample_size is None:
+        sample_size = len(continue_nodes)
+
+    new_ts = simulate(sample_size=sample_size, end_time=time, **kwargs)
+
+    new_nodes = np.where(new_ts.tables.nodes.time == time)[0]
+
+    if len(new_nodes) > len(continue_nodes):
+        raise RuntimeError(
+            f"""More lineages present in new simulation samples than in the old.. \
```

Comment:
Something more precise would be good here. This is tricky, but how about: "More lineages present at the longest-ago end of the 'continued' portion of the simulation than there are nodes in 'continue_nodes'."